File: Spring Boot/monolith/src/main/java/pl/edu/pg/benchmarking/monolith/distance/SphericalDistanceService.java
package pl.edu.pg.benchmarking.monolith.distance;

import org.springframework.stereotype.Service;

@Service
public class SphericalDistanceService {

    public double calculateDistanceBetweenPoints(Point point1, Point point2) {
        return calculateDistance(
                point1.getLatitude(), point1.getLongitude(),
                point2.getLatitude(), point2.getLongitude());
    }

    private double calculateDistance(double latitude1, double longitude1, double latitude2, double longitude2) {
        return Math.acos(
                Math.sin(Math.toRadians(latitude1)) * Math.sin(Math.toRadians(latitude2))
                        + Math.cos(Math.toRadians(latitude1)) * Math.cos(Math.toRadians(latitude2)) * Math.cos(Math.toRadians(longitude1 - longitude2))
        ) * 6371;
    }
}

File: Flask/microservices/distance-service/website/spherical_distance_service.py
import math
from website.point import Point


class SphericalDistanceService:
    def __init__(self) -> None:
        pass

    def calculate_distance(self, point: Point) -> float:
        return math.acos(math.sin(math.radians(point.latitude_1)) * math.sin(math.radians(point.latitude_2)) + \
            math.cos(math.radians(point.latitude_1)) * math.cos(math.radians(point.latitude_2)) * \
            math.cos(math.radians(point.longitude_1 - point.longitude_2))) * 6371
